### ISSUE TITLE
Alt text Missing on Item Marker Image

### ISSUE THREAD

**Issue by mdavismcd** [omeka-s-modules/Mapping#88]
Can alt text be added to the media selected as the marker image? I have the alt text module added. I also tried adding alt text manually in the media tab, but the mapping module doesn't pull it.
Example code with WAVE tool
```
<div class="mapping-marker-popup-content" style="display:none;" data-marker-id="12" data-item-id="4542">
Heading level 3
<h3>
The Carolina Inn 2014
</h3>
Missing alternative text
<img src="https://digitalprojects.davidson.edu/omeka/files/medium/304b3232c6c59d32e3df30693cb4d34196f5537e.jpg">
<div>
```
[Example site](https://digitalprojects.davidson.edu/omeka/s/encyclopedia/page/carolina-inn)

WAVE detects missing alt text when the marker isn't selected (hidden error) and when the pop-up is displaying, WAVE finds the missing alt text as well.

Thanks!

**Comment by jimsafley** [comment]
I've updated the module to automatically use the alt text as provided by the AltText module. The change will be available in the next release.

**Comment by mdavismcd** [comment]
Some of the best support both in forums and Github come from the Omeka folks! THANK YOU!

**Comment by mdavismcd** [comment]
I just updated the same line of code in my same mapping module file, cleared the cache, but I'm still not seeing the alt text of the marker image.

**Comment by jimsafley** [comment]
Try updating using the most recent commit.

**Comment by mdavismcd** [comment]
This did the trick for the marker image alt text! Thanks so much.

I'm also using the Mapping timeline integration with the [Numeric Data Types Module](https://github.com/omeka-s-modules/NumericDataTypes) and am discovering more alt text missing from the media. Two img class examples below:
`<img class="tl-timemarker-media"` 
&
`<img class="tl-media-item tl-media-image tl-media-shadow"`

Should I open up another issue on this Module?

**Comment by jimsafley** [comment]
Those images are rendered indirectly by the JS timeline library, not directly by this module's code. I'll look into how difficult it would be to inject the alt attribute into those tags, but we may have to overlook this for now.

**Comment by mdavismcd** [comment]
Understood. Thanks so much for looking into it. If I need to open an issue with the Timeline JS folks, I'm happy to do so! Thanks again!

**Comment by jimsafley** [comment]
Try the most recent commit 1ec036c462a75d87f628b8167b7d3428115cc842

**Comment by mdavismcd** [comment]
Making that one change, I get the following 
Error
Call to undefined method Omeka\Api\Representation\MediaRepresentation::altText()

Details:

Error: Call to undefined method Omeka\Api\Representation\MediaRepresentation::altText() in /home/inkandbo/digitalprojects.davidson.edu/omeka/modules/Mapping/src/Site/BlockLayout/AbstractMap.php:317
Stack trace:
0 /home/inkandbo/digitalprojects.davidson.edu/omeka/modules/Mapping/src/Site/BlockLayout/MapQuery.php(68): Mapping\Site\BlockLayout\AbstractMap->getTimelineEvent(Object(Omeka\Api\Representation\ItemRepresentation), Array, Object(Laminas\View\Renderer\PhpRenderer))
1 /home/inkandbo/digitalprojects.davidson.edu/omeka/application/src/View/Helper/BlockLayout.php(124): Mapping\Site\BlockLayout\MapQuery->render(Object(Laminas\View\Renderer\PhpRenderer), Object(Omeka\Api\Representation\SitePageBlockRepresentation))
2 /home/inkandbo/digitalprojects.davidson.edu/omeka/application/view/omeka/site/page/content.phtml(9): Omeka\View\Helper\BlockLayout->render(Object(Omeka\Api\Representation\SitePageBlockRepresentation))
3 /home/inkandbo/digitalprojects.davidson.edu/omeka/vendor/laminas/laminas-view/src/Renderer/PhpRenderer.php(505): include('/home/inkandbo/...')
4 /home/inkandbo/digitalprojects.davidson.edu/omeka/vendor/laminas/laminas-view/src/View.php(206): Laminas\View\Renderer\PhpRenderer->render(NULL)
5 /home/inkandbo/digitalprojects.davidson.edu/omeka/vendor/laminas/laminas-view/src/View.php(235): Laminas\View\View->render(Object(Laminas\View\Model\ViewModel))
6 /home/inkandbo/digitalprojects.davidson.edu/omeka/vendor/laminas/laminas-view/src/View.php(199): Laminas\View\View->renderChildren(Object(Laminas\View\Model\ViewModel))
7 /home/inkandbo/digitalprojects.davidson.edu/omeka/vendor/laminas/laminas-view/src/View.php(235): Laminas\View\View->render(Object(Laminas\View\Model\ViewModel))
8 /home/inkandbo/digitalprojects.davidson.edu/omeka/vendor/laminas/laminas-view/src/View.php(199): Laminas\View\View->renderChildren(Object(Laminas\View\Model\ViewModel))
9 /home/inkandbo/digitalprojects.davidson.edu/omeka/vendor/laminas/laminas-mvc/src/View/Http/DefaultRenderingStrategy.php(104): Laminas\View\View->render(Object(Laminas\View\Model\ViewModel))
10 /home/inkandbo/digitalprojects.davidson.edu/omeka/vendor/laminas/laminas-eventmanager/src/EventManager.php(321): Laminas\Mvc\View\Http\DefaultRenderingStrategy->render(Object(Laminas\Mvc\MvcEvent))
11 /home/inkandbo/digitalprojects.davidson.edu/omeka/vendor/laminas/laminas-eventmanager/src/EventManager.php(170): Laminas\EventManager\EventManager->triggerListeners(Object(Laminas\Mvc\MvcEvent))
12 /home/inkandbo/digitalprojects.davidson.edu/omeka/vendor/laminas/laminas-mvc/src/Application.php(366): Laminas\EventManager\EventManager->triggerEvent(Object(Laminas\Mvc\MvcEvent))
13 /home/inkandbo/digitalprojects.davidson.edu/omeka/vendor/laminas/laminas-mvc/src/Application.php(347): Laminas\Mvc\Application->completeRequest(Object(Laminas\Mvc\MvcEvent))
14 /home/inkandbo/digitalprojects.davidson.edu/omeka/index.php(21): Laminas\Mvc\Application->run()
15 {main}

**Comment by jimsafley** [comment]
This change will only work with the most recent version of Omeka S (v3.1.x). You'll need to upgrade Omeka S. The most recent version of Mapping module requires Omeka S v3.1.x, so you'll need to upgrade anyway for the module to work once we release these changes.

**Comment by mdavismcd** [comment]
Ah, that makes sense! I'm sorry to open an issue on a previous version. I greatly appreciate your help with this.
